ipl_decrypt.cpp

```diff
@@ -550,18 +551,18 @@ int decryptIPL(u8 *inData, u32 inDataSize, int version, u32 loadAddr, const char
                 if (debug) {
                     printf("decompressed %d bytes\n", decSize);
                 }
-                sprintf(szDataPath, "./F0/PSARDUMPER/stage2_%s.gz", filename);
```

Comment:
`szDataPath` is likely not large enough any more. Should also be using `snprintf` to be sure that we always null terminate as well.

Reply:
Also consider making `szDataPath` a `std::string`

Reply:
i have fixed that in the second to last commit :)

Reply:
The issue remains at some places though.
I'd suggest changing szDataPath to a `std::string` as John-K suggested, then replace
`strcpy(szDataPath, (outdir + "/F0/PSARDUMPER/12000_files_table.bin").c_str());`
with `dataPath = outdir + "/F0/PSARDUMPER/12000_files_table.bin"`
and
`sprintf(szDataPath, (outdir + "/F0/PSARDUMPER/stage3_%s").c_str(), filename);`
with `dataPath = outdir + "/F0/PSARDUMPER/stage3_" + std::string(filename)` (hoping that works as I hope it does)

Reply:
i think it should be fixed on the other commits i've made just now